compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -561,6 +561,17 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
             false
         }
         compareTypeBounds
+      case tp2: AppliedTermRef =>
```

Comment:
Any reason to have this in thirdTry and not firstTry? Also I think it'd be good to have more testcases that stress-test the subtype checks for AppliedTermRef. For example, what happens when an AppliedTermRef is hidden in a type alias?

Reply:
We do have a test that relies on a type application:
```scala
val t: Id[{a + 1}] = a + 1
```
I'll also add one that relies on a simple alias.